libp2p/tools/pubsub/floodsub_integration_test_settings.py

```diff
@@ -180,59 +188,45 @@
     node_map = {}
     pubsub_map = {}
 
-    async def add_node(node_id_str: str) -> None:
+    async def add_node(node_id_str: str):
         pubsub_router = router_factory(protocols=obj["supported_protocols"])
         pubsub = PubsubFactory(router=pubsub_router)
         await pubsub.host.get_network().listen(LISTEN_MADDR)
         node_map[node_id_str] = pubsub.host
         pubsub_map[node_id_str] = pubsub
 
-    tasks_connect = []
-    for start_node_id in adj_list:
-        # Create node if node does not yet exist
-        if start_node_id not in node_map:
-            await add_node(start_node_id)
+    all_node_ids = _collect_node_ids(adj_list)
+
+    for node in all_node_ids:
+        await add_node(node)
+
+    connect_tasks = []
+    for node, neighbors in adj_list.items():
+        for neighbor_id in neighbors:
+            await connect(node_map[node], node_map[neighbor_id])
 
-        # For each neighbor of start_node, create if does not yet exist,
-        # then connect start_node to neighbor
-        for neighbor_id in adj_list[start_node_id]:
-            # Create neighbor if neighbor does not yet exist
-            if neighbor_id not in node_map:
-                await add_node(neighbor_id)
-            tasks_connect.append(
-                connect(node_map[start_node_id], node_map[neighbor_id])
-            )
-    # Connect nodes and wait at least for 2 seconds
-    await asyncio.gather(*tasks_connect, asyncio.sleep(2))
+    # NOTE: the test using this routine will fail w/o these sleeps...
```

Comment:
had some trouble w/ this suite of tests under python 3.6.... seems like something changed behind the scenes w/ `gather`.... ultimately tried to streamline the implementation and found that the `sleep`s are important...

given that we are moving towards a `trio` refactor i don't think its worth the resources right now to debug further....

Reply:
Maybe add a tracking issue to revisit this after switching to trio?